pkg/kinflate/resource/configmap.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	corev1 "k8s.io/api/core/v1"
+	manifest "k8s.io/kubectl/pkg/apis/manifest/v1alpha1"
+	cutil "k8s.io/kubectl/pkg/kinflate/configmapandsecret/util"
+	"k8s.io/kubectl/pkg/kinflate/hash"
+)
+
+// (Note): pass in loader which has rootPath context and knows how to load
+// files given a relative path.
+// NewFromConfigMap returns a Resource given a configmap metadata from manifest
+// file.
+func NewFromConfigMap(cm manifest.ConfigMap) (*Resource, error) {
+	corev1CM, err := makeConfigMap(cm)
+	if err != nil {
+		return nil, err
+	}
+	h, err := hash.ConfigMapHash(corev1CM)
+	if err != nil {
+		return nil, err
+	}
+	data, err := objectToUnstructured(corev1CM)
+	if err != nil {
+		return nil, err
+	}
+	return &Resource{
+		Data:       data,
+		NameSuffix: h,
```

Comment:
Do we need this? Can't we have the name already include the suffix?

Reply:
Discussed with @mengqiy , we need the old-name of the resources because we need to update references for old-name after transforming.

So may be we can call OldName which is applicable to all the resources anyways and update the name in the Data.